legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -30,6 +32,44 @@ def validate(business, incorporation_json):
     if business.identifier != temp_identifier:
         msg.append({'error': babel('Business Identifier does not match the identifier in filing.')})
 
+    err = validate_offices(incorporation_json)
+    if err:
+        msg.append(err.msg)
+
+    if msg:
+        return Error(HTTPStatus.BAD_REQUEST, msg)
+    return None
+
+
+def validate_offices(incorporation_json) -> Error:
+    """Validate the office addresses of the incorporation filing."""
+    offices_array = incorporation_json['filing']['incorporationApplication']['offices']
+    addresses = offices_array
+    msg = []
+
+    for item in addresses.keys():
+        for k, v in addresses[item].items():
+            region = v['addressRegion']
+            country = v['addressCountry']
+
+            if region != 'BC':
+                path = '/filing/incorporationApplication/offices/%s/%s/addressRegion' % (
+                    item, k
+                )
+                msg.append({'error': "Address Region must be 'BC'.",
+                            'path': path})
+
+            try:
+                country = pycountry.countries.search_fuzzy(country)[0].alpha_2
+                if country != 'CA':
+                    raise LookupError
+            except LookupError:
+                err_path = '/filing/incorporationApplication/offices/%s/%s/addressCountry' % (
+                    item, k
+                )
+                msg.append({'error': "Address Country must be 'CA'.",
+                            'path': err_path})
     if msg:
         return Error(HTTPStatus.BAD_REQUEST, msg)
+
```

Comment:
I think it would be better to return just the list of messages. The actual message code from this response doesn't get used in the calling validate() function, only the messages, so might as well just return the list of messages ("msg") and adjust accordingly in lines 35-37. 

Your choice - it was jarring, though, to see the resturned Error object from this function though, so it might confuse others also.

Reply:
Sounds good, will update!